test/lib/ufe/testUIIcons.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env mayapy
+#
+# Copyright 2022 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from maya import cmds
+import maya.OpenMayaUI as mui
+
+import fixturesUtils
+
+import unittest
+
+import ufe
+import mayaUsd
+from pxr import Usd
+
+from shiboken2 import wrapInstance
+from PySide2.QtGui import QIcon
+
+class UIIconsTestCase(unittest.TestCase):
+    """
+    Tests that all prims have the icon file we expect.
+    - Either directly from the UIInfoHandler
+    - Or from the ancestor node types
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        inputPath = fixturesUtils.readOnlySetUpClass(__file__, initializeStandalone=False)
+
+    def setUp(self):
+        # Start with just empty stage
+        cmds.file(force=True, new=True)
+        import mayaUsd_createStageWithNewLayer
+        self.proxyShapePathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        self.assertIsNotNone(self.proxyShapePathStr)
+        self.stage = mayaUsd.lib.GetPrim(self.proxyShapePathStr).GetStage()
+        self.assertIsNotNone(self.stage)
+
+        rid = ufe.RunTimeMgr.instance().getId('USD')
+        self.ufeUIInfo = ufe.UIInfoHandler.uiInfoHandler(rid)
+        self.assertIsNotNone(self.ufeUIInfo)
+        self.baseIconName = 'out_' + ufe.RunTimeMgr.instance().getName(rid) + '_';
+
+    @classmethod
+    def tearDownClass(cls):
+        pass
+
+    def _createIconFromFile(self, iconName):
+        swigIcon = mui.MQtUtil.createIcon(iconName)
+        if swigIcon:
+            return wrapInstance(int(swigIcon), QIcon)
+        return None
```

Comment:
Simple helper to try and create an icon from the input filename. This uses the MQtUtil which is why this tests needs to be interactive. We do this to mimic what happens in Maya (searching thru the XBMLANGPATH env var for a matching file).